From 620a20af11ce016873f9bf3c7b3cdbdf8e39eb42 Mon Sep 17 00:00:00 2001
From: Gwenn Bourree <gwenn.bourree@intel.com>
Date: Wed, 1 Oct 2014 17:09:36 +0200
Subject: [PATCH 1/1] [Telephony/MDM_CTRL]: solve compile error for kernel-dev

BZ: GMINL-1875

In order to be compilant with the kernel-dev (3.17-rc7)
where some changes have been done in the GPIO header
files, some modifications are required in modem control
driver. Moreover, the modem control driver can only work
if the INTEL_SOC_PMIC driver exists in the configuration.

Change-Id: Ia2f89e404b1a5b421593e22e0fbb0eb34280d9d4
Signed-off-by: Gwenn Bourree <gwenn.bourree@intel.com>
---
 drivers/staging/modem_control/Kconfig    | 1 +
 drivers/staging/modem_control/mdm_util.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/staging/modem_control/Kconfig b/drivers/staging/modem_control/Kconfig
index 83218bd01a70..895b03fb5e1a 100644
--- a/drivers/staging/modem_control/Kconfig
+++ b/drivers/staging/modem_control/Kconfig
@@ -5,6 +5,7 @@
 
 config MDM_CTRL
 	tristate "Modem control driver"
+	depends on INTEL_SOC_PMIC
 	default y
 	---help---
 	  The modem control driver for IMC modems. This driver will perform any
diff --git a/drivers/staging/modem_control/mdm_util.h b/drivers/staging/modem_control/mdm_util.h
index 575f442f8be9..01d85359b982 100644
--- a/drivers/staging/modem_control/mdm_util.h
+++ b/drivers/staging/modem_control/mdm_util.h
@@ -33,6 +33,7 @@
 #include <linux/interrupt.h>
 #include <linux/irq.h>
 #include <linux/gpio.h>
+#include <linux/gpio/consumer.h>
 #include <linux/mutex.h>
 #include <linux/wait.h>
 #include <linux/delay.h>
-- 
1.9.1

